Source/SwiftLintFramework/Rules/Idiomatic/NoConstantConditionRule.swift

```diff
@@ -0,0 +1,566 @@
+// swiftlint:disable file_length cyclomatic_complexity
+import SwiftSyntax
+
+struct NoConstantConditionRule: ConfigurationProviderRule, SwiftSyntaxRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static let description = RuleDescription(
+        identifier: "no_constant_condtion",
+        name: "No constant condition",
+        description: "Disallow branching statements when the branching condition is constant",
```

Comment:
Style for the description feels a tiny bit off (for example, none start with "Disallow"). Maybe something like "Branching statements with constant conditions should be avoided"

Reply:
Thanks. I’ve used the wording from [`ESLint`](https://eslint.org/docs/latest/rules/no-constant-condition) as an inspiration, but I’m ok with your suggestion as well.